science/nektarpp/Portfile

```diff
@@ -0,0 +1,160 @@
+                        sha256  7d74d7efbe35dc5a497bd179bdc87b4670740e466f67043d41e0b0546ce76cb5 \
+                        size    130331
+
+# Only the full Nektar++ source should be extracted - CMake will extract any
+# third-party dependencies.
+extract.only            nektar-5.0.0.tar.bz2
+
+patchfiles              no-homebrew.patch \
+                        avoid-python-install.patch \
+                        thirdparty-sources.patch
+
+compiler.cxx_standard   2011
+cmake.build_type        Release
+
+mpi.setup
+
+depends_lib-append      port:arpack \
+                        port:boost \
+                        port:fftw-3 \
+                        port:opencascade \
+                        port:scotch \
+                        port:tinyxml \
+                        port:zlib
+
+# Don't build demos and tests by default.
+configure.args-append   -DNEKTAR_BUILD_DEMOS=OFF \
+                        -DNEKTAR_BUILD_TESTS=OFF \
+                        -DNEKTAR_BUILD_UNIT_TESTS=OFF \
+                        -DNEKTAR_SOLVER_DIFFUSION=OFF \
+                        -DNEKTAR_SOLVER_DUMMY=OFF \
+                        -DNEKTAR_SOLVER_ELASTICITY=OFF \
+                        -DNEKTAR_SOLVER_MMF=OFF \
+                        -DNEKTAR_USE_ARPACK=ON \
+                        -DNEKTAR_USE_FFTW=ON \
+                        -DNEKTAR_USE_MESHGEN=ON \
+                        -DNEKTAR_USE_SCOTCH=ON
+
+variant demos description {
+    Build additional Nektar++ demos
+} {
+    configure.args-replace \
+                        -DNEKTAR_BUILD_DEMOS=OFF -DNEKTAR_BUILD_DEMOS=ON
+}
+
+variant petsc description {
+    Enable PETSc support for linear algebra solvers.
+} {
+    depends_lib-append port:petsc
+
+    configure.args-append \
+                        -DNEKTAR_USE_PETSC=ON
+}
+
+if {[mpi_variant_isset]} {
+    distfiles-append    gsmpi-1.2.1_1.tar.bz2
+    checksums-append    gsmpi-1.2.1_1.tar.bz2 \
+                        rmd160 3a1ad71c552ea9d62bebabd4ed528531f9c0eec6 \
+                        sha256 c9bc966c92f45e81f4e9d1887dd7c59808e7a4ef9ec41e1e5f1e6c5c01745211 \
+                        size 39737
+
+    depends_lib-append port:hdf5
+
+    require_active_variants hdf5 [mpi_variant_name]
+    require_active_variants scotch [mpi_variant_name]
+
+    configure.args-append \
+                        -DNEKTAR_USE_HDF5=ON \
+                        -DNEKTAR_USE_MPI=ON
+}
+
+variant oce description {
+    use Community Edition version of OpenCASCADE
+} {
+    depends_lib-replace port:opencascade port:oce
+}
+
+post-extract {
+    # Copy additional third-party sources into correct position.
+    file mkdir ${worksrcpath}/ThirdParty
+    copy ${distpath}/triangle-1.6.zip ${worksrcpath}/ThirdParty/triangle-1.6.zip
+    copy ${distpath}/tetgen-1.5.zip ${worksrcpath}/ThirdParty/tetgen-1.5.zip
+
+    if {[mpi_variant_isset]} {
+        copy ${distpath}/gsmpi-1.2.1_1.tar.bz2 ${worksrcpath}/ThirdParty/gsmpi-1.2.1_1.tar.bz2
+    }
```

Comment:
To avoid repeating version numbers here, you could programmatically copy all files except the main file:

```
    foreach distfile ${distfiles} {
        set distfile [lindex [split ${distfile} :] 0]
        if {${distfile} ne ${main_distfile}} {
            copy ${distpath}/${distfile} ${worksrcpath}/ThirdParty/
        }
    }
```

Reply:
Seems to work well, thanks!